specs/error_state/ErrorOutOfGasSHA3.md

```diff
@@ -0,0 +1,30 @@
+# ErrorOutOfGasSHA3 state for SHA3 OOG error
+
+## Procedure
+
+Handle the corresponding out of gas error for `SHA3`.
+
+### EVM behavior
+
+For `SHA3`, the out of gas error may occur in conditions:
+
+1. `memory_offset + memory_length > 0x1FFFFFFFE0 ((2^32 - 1) * 32)` if `memory_length` is non-zero.
+2. Gas left is less than a constant gas cost of `30`.
+3. Gas left is less than the sum of both constant and dynamic gas costs. The dynamic gas cost is calculated for memory expansion and copying (variable depending on the memory size copied to memory).
+
+### Constraints
+
+1. Constrain either `memory_offset + memory_length > 0x1FFFFFFFE0` or `gas_left < gas_cost` must be true.
```

Comment:
this is checking `ErrGasUintOverflow` ?

Reply:
Added a comment in upstream `ErrGasUintOverflow` PR https://github.com/privacy-scaling-explorations/zkevm-specs/pull/361#issuecomment-1478969271